### ISSUE TITLE
Let DataTable accept autoLayout and scrollable props at the same time

### ISSUE THREAD

**Issue by hstevanoski** [primefaces/primereact#2230]
**I'm submitting a ...** 
```
[ ] bug report
[x] feature request
[ ] support request
```
Currently, the DataTable component doesn't support this feature and if both are set, it ignores the `scrollable` prop.
In the company that I work in, due to the nature of the product, we depend heavily on DataTables and each contains around 15-30 columns, which means that all our tables have to be scrollable.
This being the case, we used to write custom widths for the columns (using `style` and `bodyStyle` Column props). However, we realized that this doesn't work with translation very well - a small sentence written in one language usually turns out to be a larger sentence in another. So, as a result, after translating, we end up with cases where most of our table headers and some of our table cells span two lines (as there's not enough width for the translated content) - or the opposite, there's ample of space and we end up wasting space.
I think the solution to this is to make the DataTable component support `autoLayout` + `scrollable` at the same time (or to support a completely new prop, whatever's easier for implementation).
I reckon this will solve issues related to unpredictable content length too (I understand that this is not a common case scenario, but we do have such cases in our application).
Because there are always edge cases and primereact users might want to have control over the widths when both these props are applied, I think we would need controlling minimum and maximum column widths as well (I am not sure if it is feasible to respect the `style`/`bodyStyle` props - that's up to you).
To sum up - both the lengths of column titles and table cells should be respected and widths of both should be set according to the max value of both.

* **React version:**
v17.0.2

* **PrimeReact version:**
v6.5.0

* **Browser:** all
 
* **Language:** all

**Comment by mertsincan** [comment]
Hi @hstevanoski,

DataTable has been reimplemented on 7.0. Could you please try it with the latest version?

**Comment by melloware** [comment]
No response from OP so closing issue
